src/sentry/remote_config/storage.py

```diff
@@ -105,5 +95,65 @@ def _make_storage_config(self) -> dict | None:
             return None
 
 
-def make_storage(key: Project):
-    return StorageBackend(key)
+class ConfigurationBackend:
+    def __init__(self, project: Project) -> None:
+        self.project = project
+        self.key = f"configurations/{self.project.id}/production"
+
+        self.cache = ConfigurationCache(self.key)
+        self.storage = ConfigurationStorage(self.key)
+
+    def get(self) -> StorageFormat | None:
+        cache_result = self.cache.get()
+        if cache_result is not None:
+            return cache_result
+
+        storage_result = self.storage.get()
+        self.cache.set(storage_result)
+        return storage_result
+
+    def set(self, value: StorageFormat) -> None:
+        self.cache.set(value)
+        self.storage.set(value)
+
+    def pop(self) -> None:
+        self.cache.pop()
+        self.storage.pop()
+
+
+class APIBackendDecorator:
```

Comment:
Don't think this is technically a decorator but could be confusing / someone might try to use it that way.. can we just call it APIBackend?

Reply:
Its a decorator.  See: https://en.wikipedia.org/wiki/Decorator_pattern

Reply:
Ah gotcha, I was thinking about python decorators with the @ syntax.